src/components/Navbar.jsx

```diff
@@ -0,0 +1,77 @@
+// src/components/Navbar.js
+import { useState } from 'react';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+
+const Navbar = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const router = useRouter();
+
+  const getLinkClass = (path) =>
+    router.pathname === path
+      ? 'py-4 px-2 text-indigo-500 border-b-4 border-indigo-500 font-semibold'
+      : 'py-4 px-2 text-gray-500 border-b-4 border-transparent hover:border-indigo-500 transition duration-300';
+
+  return (
+    <nav className="bg-black shadow-lg">
+      <div className="max-w-6xl mx-auto px-4">
+        <div className="flex justify-between">
+          <div className="flex space-x-7">
+            <div>
+              <Link href="/" legacyBehavior>
+                <a className="flex items-center py-4 px-2">
+                  <span className="font-semibold text-gray-500 text-lg">RK</span>
+                </a>
+              </Link>
+            </div>
+            <div className="hidden md:flex items-center space-x-1">
+              <Link href="/" legacyBehavior>
+                <a className={getLinkClass('/')}>Home</a>
+              </Link>
+              <Link href="/about" legacyBehavior>
+                <a className={getLinkClass('/about')}>About</a>
+              </Link>
+              <Link href="/services" legacyBehavior>
+                <a className={getLinkClass('/services')}>Services</a>
+              </Link>
+              <Link href="/contact" legacyBehavior>
+                <a className={getLinkClass('/contact')}>Contact</a>
+              </Link>
+            </div>
+          </div>
+          <div className="md:hidden flex items-center">
+            <button className="outline-none mobile-menu-button" onClick={() => setIsOpen(!isOpen)}>
```

Comment:
**Issue**: The onClick handler directly modifies the state based on the previous state without using the prevState argument in the updater function. This can lead to state inconsistency due to the asynchronous nature of state updates. <br> **Fix**: Use the functional update form of setState that accepts the previous state as an argument to ensure the state update is always consistent. <br> **Code Suggestion**: 
 ```
 - <button className="outline-none mobile-menu-button" onClick={() => setIsOpen(!isOpen)}>
 + <button className="outline-none mobile-menu-button" onClick={() => setIsOpen(prevState => !prevState)}>
 ```